src/stan/mcmc/hmc/hamiltonians/diag_e_metric.hpp

```diff
@@ -48,6 +47,8 @@ class diag_e_metric : public base_hamiltonian<Model, diag_e_point, BaseRNG> {
     for (int i = 0; i < z.p.size(); ++i)
       z.p(i) = rand_diag_gaus() / sqrt(z.inv_e_metric_(i));
```

Comment:
We need to only initialize random momentums for non-masked variables.

Reply:
If the masked variables are randomly initialized to a non-zero momentum, they might move, or some energy calculation might be off. So want to make sure and zero them.